### PR TITLE
Regenerate compute_gapic.yaml

### PR DESCRIPTION
**Pull request by andreamlin**
Regenerate the compute config using artman.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: gapic/google/compute/v1/compute_gapic.yaml [hunk 1]
```diff
@@ -7515,6 +7517,416 @@ interfaces:
...
+  methods:
+  - name: compute.regionDisks.createSnapshot
+    # TODO: Configure which groups of fields should be flattened into method
+    # params.
+    flattening:
+      groups:
+      - parameters:
+        - disk
+        - requestId
+        - snapshotResource
```
Comment (pongad) [inline gapic/google/compute/v1/compute_gapic.yaml]: What is this? I don't see it in the discovery doc; is it the request body?

File: gapic/google/compute/v1/compute_gapic.yaml [hunk 2]
```diff
@@ -7515,6 +7517,416 @@ interfaces:
...
+    # TODO: Configure which fields are required.
+    required_fields:
+    - disk
+    - requestId
```
Comment (pongad) [inline gapic/google/compute/v1/compute_gapic.yaml]: `requestId` is optional.

File: gapic/google/compute/v1/compute_gapic.yaml [hunk 3]
```diff
@@ -7515,6 +7517,416 @@ interfaces:
...
+        - requestId
+        - sourceImage
+        - region
+        - diskResource
+    # TODO: Configure which fields are required.
+    required_fields:
+    - requestId
+    - sourceImage
+    - region
+    - diskResource
```
Comment (pongad) [inline gapic/google/compute/v1/compute_gapic.yaml]: requestId and sourceImage are optional. This seems to be happening in other places as well.

Is "region" here the whole "projects/P/regions/R"?

File: gapic/google/compute/v1/compute_gapic.yaml (hunk 1 above)
Comment (andreamlin) [reply gapic/google/compute/v1/compute_gapic.yaml]: gapic-generator automatically appends `Resource` to the variable name of a request body object, in this case `snapshot`.

File: gapic/google/compute/v1/compute_gapic.yaml (hunk 2 above)
Comment (andreamlin) [reply gapic/google/compute/v1/compute_gapic.yaml]: Hmm this sounds like gapic-generator needs to be changed to exclude query parameters from the required_fields list config. I will make a PR for that first, which blocks this one.

File: gapic/google/compute/v1/compute_gapic.yaml (hunk 3 above)
Comment (andreamlin) [reply gapic/google/compute/v1/compute_gapic.yaml]: Will make query params optional in a bit.

re: "region", yes. See line 7829 for the mapping of the `region` param to its field_name_pattern `projectRegion`, which has a path template defined in lines 7532-7533.

File: gapic/google/compute/v1/compute_gapic.yaml (hunk 1 above)
Comment (pongad) [reply gapic/google/compute/v1/compute_gapic.yaml]: Makes sense

File: gapic/google/compute/v1/compute_gapic.yaml (hunk 3 above)
Comment (pongad) [reply gapic/google/compute/v1/compute_gapic.yaml]: Right makes sense too

**Comment by pongad** [comment]
I'm good with this, assuming we make query params optional at some point.

**Pull request merged**